helion/_compiler/type_propagation.py

```diff
@@ -710,6 +710,10 @@ def propagate_call(
                 return ExtendedAST.current()[-1]._type_info
             assert fn._type_function is not None
             return fn._type_function(*args, **kwargs, origin=origin)
+
+        if self.value is print:
+            return NoType(origin=origin)
```

Comment:
Rather than special casing print, perhaps we could do:
```
_function_replacements : {
  print: hl.device_print
}
```
and here do:
```
fn = _function_replacements.get(self.value, self.value)
```